partner-chains-cli/src/deregister/tests.rs

```diff
@@ -0,0 +1,246 @@
+use crate::config::{CHAIN_CONFIG_FILE_PATH, RESOURCES_CONFIG_FILE_PATH};
+use crate::deregister::DeregisterCmd;
+use crate::pc_contracts_cli_resources::tests::establish_pc_contracts_cli_configuration_io;
+use crate::pc_contracts_cli_resources::PcContractsCliResources;
+use crate::tests::{MockIO, MockIOContext};
+use crate::CmdRun;
+use serde_json::json;
+
+const MY_PAYMEMENT_SKEY: &str = "my_payment.skey";
+const MY_COLD_VKEY: &str = "my_cold.vkey";
+
+#[test]
+fn happy_path() {
+	let mock_context = MockIOContext::new()
```

Comment:
*Maybe* the mock setup code could be moved out of the individual test functions? It makes it a bit harder to read the test and see what the behavior and expectation is.

So the happy_path() would be like
```Rust
fn happy_path() {
  let mock_context = build_mock_context_for_happy_path();
  let res = DeregisterCmd.run(&mock_context);
  assert!(res.is_ok());
}
```
Obviously the mock code doesn't disappear but it makes the relevant test code more visible I think

Reply:
In this case I prefer to have explicit "given" without a more layers of indirection. I think irrelevant details, like 5 IO operations to read Cardano key file are abstracted and it is enough.